bundles/org.eclipse.jface/src/org/eclipse/jface/preference/SpinnerFieldEditor.java

```diff
@@ -0,0 +1,370 @@
+	 */
+	private int oldValue;
+
+	/**
+	 * Value that will feed {@link Spinner#setPageIncrement(int)}.
+	 */
+	private int pageIncrementValue = 1;
+
+	/**
+	 * Value that will feed {@link Spinner#setTextLimit(int)}.
+	 */
+	private int textLimit = Spinner.LIMIT;
+
+	/**
+	 * The spinner, or <code>null</code> if none.
+	 */
+	protected Spinner spinner;
+
+	/**
+	 * Creates a spinner field editor using default values.
+	 *
+	 * @param name      the name of the preference this field editor works on
+	 * @param labelText the label text of the field editor
+	 * @param parent    the parent of the field editor's control
+	 */
+	public SpinnerFieldEditor(final String name, final String labelText, final Composite parent) {
+		super(name, labelText, parent);
+		updateSpinner();
+	}
+
+	/**
+	 * Creates a spinner field editor with particular spinner values.
+	 *
+	 * @param name      the name of the preference this field editor works on
+	 * @param labelText the label text of the field editor
+	 * @param parent    the parent of the field editor's control
+	 * @param min       the value used for {@link Spinner#setMinimum(int)}.
+	 * @param max       the value used for {@link Spinner#setMaximum(int)}.
+	 */
+	public SpinnerFieldEditor(final String name, final String labelText, final Composite parent, final int min,
+			final int max) {
+		this(name, labelText, parent);
+		setMinimum(min);
+		setMaximum(max);
+	}
+
+	@Override
+	protected void adjustForNumColumns(final int numColumns) {
+		((GridData) spinner.getLayoutData()).horizontalSpan = numColumns - 1;
+	}
+
+	@Override
+	protected void doFillIntoGrid(final Composite parent, final int numColumns) {
+		getLabelControl(parent).setLayoutData(new GridData());
+
+		spinner = getSpinnerControl(parent);
+		final var gd = new GridData(GridData.FILL_HORIZONTAL);
+		gd.verticalAlignment = GridData.FILL;
+		gd.horizontalSpan = numColumns - 1;
+		gd.grabExcessHorizontalSpace = true;
+		spinner.setLayoutData(gd);
+	}
+
+	@Override
+	protected void doLoad() {
+		if (spinner != null) {
+			final int value = getPreferenceStore().getInt(getPreferenceName());
+			spinner.setSelection(value);
+			oldValue = value;
+		}
+	}
+
+	@Override
+	protected void doLoadDefault() {
+		if (spinner != null) {
+			final int value = getPreferenceStore().getDefaultInt(getPreferenceName());
+			spinner.setSelection(value);
+		}
+		valueChanged();
+	}
+
+	@Override
+	protected void doStore() {
+		getPreferenceStore().setValue(getPreferenceName(), spinner.getSelection());
+	}
+
+	/**
+	 * Returns the value that will be used for {@link Spinner#setDigits(int)}.
+	 *
+	 * @return the value (default 0).
+	 * @see Spinner#setDigits(int)
+	 */
+	public int getDigits() {
+		return digits;
+	}
+
+	/**
+	 * Returns the value that will be used for {@link Spinner#setIncrement(int)}.
+	 *
+	 * @return the value (default 1).
+	 * @see Spinner#setIncrement(int)
+	 */
+	public int getIncrement() {
+		return incrementValue;
+	}
+
+	/**
```

Comment:
A lot of these javadoc comments seem redundant. In particular, I am talking about comments referring to the setter of a getter.

```suggestion
	/** {@return the maximum value this spinner may take on (10 by default) }
```